src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/scalable/TrainVariantAnnotationModel.java

```diff
@@ -0,0 +1,260 @@
+            }
+            logger.info(String.format("Scores written to %s.", outputScoresFile.getAbsolutePath()));
+        }
+
+        logger.info("Training complete.");
+
+        // TODO some duplication of code here and in scoring tool
+        try (final HDF5File outputScoresHDF5File = new HDF5File(outputScoresFile, HDF5File.OpenMode.READ_ONLY);
+             final HDF5File inputAnnotationsHDF5File = new HDF5File(this.inputAnnotationsHDF5File, HDF5File.OpenMode.READ_ONLY)) {
+            IOUtils.canReadFile(outputScoresHDF5File.getFile());
+            IOUtils.canReadFile(inputAnnotationsHDF5File.getFile());
+            final List<Double> scores = Doubles.asList(outputScoresHDF5File.readDoubleArray("/data/scores"));
+            final List<Boolean> isBiallelicSNP = readBooleanList(inputAnnotationsHDF5File, "/data/is_biallelic_snp");
+            final List<Boolean> isTransition = readBooleanList(inputAnnotationsHDF5File, "/data/is_transition");
+            final List<Boolean> isTruth = readBooleanList(inputAnnotationsHDF5File, "/data/is_truth");
+
+            // Find the score cutoff values which correspond to the various tranches of calls requested by the user
+            final int nCallsAtTruth = TruthSensitivityTranche.countCallsAtTruth(scores, isTruth, Double.NEGATIVE_INFINITY);
+            final TruthSensitivityTranche.TruthSensitivityMetric metric = new TruthSensitivityTranche.TruthSensitivityMetric(nCallsAtTruth);
+            final List<TruthSensitivityTranche> tranches = TruthSensitivityTranche.findTranches(scores, isBiallelicSNP, isTransition, isTruth, truthSensitivityTranches, metric, mode);
+            tranchesStream.print(TruthSensitivityTranche.printHeader());
+            tranchesStream.print(TruthSensitivityTranche.tranchesString(tranches));
+        } catch (final HDF5LibException exception) {
+            throw new GATKException(String.format("Exception encountered during reading of scores from %s or annotations from %s: %s",
+                    outputScoresFile.getAbsolutePath(), inputAnnotationsHDF5File.getAbsolutePath(), exception));
+        }
+
+        logger.info(String.format("%s complete.", getClass().getSimpleName()));
+
+        return null;
+    }
+
+    private static List<Boolean> readBooleanList(final HDF5File annotationsHDF5File,
+                                                 final String path) {
+        return Arrays.stream(annotationsHDF5File.readDoubleArray(path))
+                .mapToObj(d -> (d == 1))
+                .collect(Collectors.toList());
+    }
+
+    private static List<String> composePythonArguments(final File rawAnnotationsFile,
+                                                       final File hyperparametersJSONFile,
+                                                       final String outputPrefix) {
+        try {
+            return new ArrayList<>(Arrays.asList(
+                    "--raw_annotations_file=" + rawAnnotationsFile.getCanonicalPath(),
```

Comment:
Introduce constants.

Reply:
I think we do it this way in the gCNV code, which makes it easier to take a quick glance and understand what the python command line looks like. OTOH we might indeed want to extract some common train/score arguments, as you say elsewhere.